std/experimental/color/hsx.d

```diff
@@ -0,0 +1,443 @@
+{
+@safe pure nothrow @nogc:
+
+    static if (isFloatingPoint!ComponentType_)
+    {
+        /** Type of the hue components. */
+        alias HueType = ComponentType_;
+        /** Type of the s and x components. */
+        alias ComponentType = ComponentType_;
+    }
+    else
+    {
+        /** Type of the hue components. */
+        alias HueType = ComponentType_;
+        /** Type of the s and x components. */
+        alias ComponentType = NormalizedInt!ComponentType_;
+    }
+
+    /** The parent RGB color space. */
+    enum colorSpace = colorSpace_;
+    /** The parent RGB color space descriptor. */
+    enum RGBColorSpaceDesc!F colorSpaceDesc(F = double) = rgbColorSpaceDef!F(colorSpace_);
+    /** The color type from the HSx family. */
+    enum type = type_;
+
+    // mixin the color channels according to the type
+    mixin("HueType " ~ Components!type[0] ~ " = 0;");
+    mixin("ComponentType " ~ Components!type[1] ~ " = 0;");
+    mixin("ComponentType " ~ Components!type[2] ~ " = 0;");
+
+    /** Get hue angle in degrees. */
+    @property double degrees() const
+    {
+        static if (!isFloatingPoint!ComponentType_)
+            return h * (360/(ComponentType_.max + 1.0));
+        else
+            return (h < 0 ? 1 - h%1 : h%1) * 360;
+    }
+    /** Set hue angle in degrees. */
+    @property void degrees(double angle)
+    {
+        static if (!isFloatingPoint!ComponentType_)
+            h = cast(ComponentType_)(angle * ((ComponentType_.max + 1.0)/360));
+        else
+            h = angle * 1.0/360;
+    }
+
+    /** Get hue angle in radians. */
+    @property double radians() const
+    {
+        static if (!isFloatingPoint!ComponentType_)
+            return h * ((PI*2)/(ComponentType_.max + 1.0));
+        else
+            return (h < 0 ? 1 - h%1 : h%1) * (PI*2);
+    }
+    /** Set hue angle in radians. */
+    @property void radians(double angle)
+    {
+        static if (!isFloatingPoint!ComponentType_)
+            h = cast(ComponentType_)(angle * ((ComponentType_.max + 1.0)/(PI*2)));
+        else
+            h = angle * 1.0/(PI*2);
+    }
+
+    /** Construct a color from hsx components. */
+    this(HueType h, ComponentType s, ComponentType x)
+    {
+        mixin("this." ~ Components!type[0] ~ " = h;");
+        mixin("this." ~ Components!type[1] ~ " = s;");
+        mixin("this." ~ Components!type[2] ~ " = x;");
+    }
+
+    static if (!isFloatingPoint!ComponentType_)
+    {
+        /** Construct a color from hsx components. */
+        this(HueType h, ComponentType.IntType s, ComponentType.IntType x)
+        {
+            mixin("this." ~ Components!type[0] ~ " = h;");
+            mixin("this." ~ Components!type[1] ~ " = ComponentType(s);");
+            mixin("this." ~ Components!type[2] ~ " = ComponentType(x);");
+        }
+    }
+
+    /** Cast to other color types */
```

Comment:
I suggest to add a second paragraph with something like this (but better written...).

_This cast is syntactic sugar since it forwards convertColor. It is not related to the langauge built-in casts thus it's completly safe_.

Ditto for the other _opCast()_ of the package.